Fixed nrnivmodl-core invocation

ohm314 opened the pull request:
For coreneuron@develop the -c argument to `nrnivmodl-core` was dropped.
This is to reflect this change here.

One thing that we might have to look at: In the coreneuron package a `develop` as well as a `master` version is declared. They both would probably point to the HEAD of master of the git repo. Does this mean that packages that depend on `coreneuron@master` should also be updated to remove the `-c` from their `nrnivmodl-core` calls?

Commit:
Fixed nrnivmodl-core invocation

For coreneuron@develop the -c argument to nrnivmodl-core was dropped.
This is to reflect this change here.

jorblancoa commented:
As far as I understand the master version is kept for compatibility with the old neurodamus package.
So I guess while this master version is still present we should remove the '-c' argument in that case also.

Commit:
Fix thanks to @jorblancoa's comment

pramodk commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -70,9 +70,15 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
-        which('nrnivmodl-core')(
-            '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
-            '-v', str(spec.version), '-c', mods_location)
+        if (str(spec['coreneuron'].version) == 'develop') or (str(spec['coreneuron'].version) ==
```
we should use `spec.satisfies('coeneuron@develop')` instead. And I would remove master version.

ohm314 replied:
@jorblancoa suggested to add the check on master version. Are you saying I should revert back?

pramodk replied:
hmm..I see the master version referenced in package recipe (which should be removed soon). You can take care of comparing version by using `satisfies` and then we can merge.

Commit:
Fix thanks to @pramodk's suggestion

iomaganaris commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -70,9 +70,14 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
-        which('nrnivmodl-core')(
-            '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
-            '-v', str(spec.version), '-c', mods_location)
+        if spec.satisfies('coeneuron@develop') or spec.satisfies('coeneuron@master'):
```
I was using this branch and I came across a small issue. I think this should be:
```suggestion
        if spec.satisfies('^coreneuron@develop') or spec.satisfies('^coreneuron@master'):
```

ohm314 replied:
You're right!

iomaganaris committed:
Various small fixes

- Removed -v and -n arguments from nrnivmodl-core with latest coreneuron
  branches
- Removed checking for suffix `self.mech_name` in generated libcorenrnmech.*
- Small typos fixes

pramodk commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -70,11 +70,15 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
-        which('nrnivmodl-core')(
-            '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
-            '-v', str(spec.version), '-c', mods_location)
+        if spec.satisfies('^coreneuron@develop') or spec.satisfies('^coreneuron@master'):
```
@iomaganaris : I think we should do opposite logic here i.e. if version is <= 0.16 (latest release) then use old logic else do new options. This way when there will be new release, it will work as well.

iomaganaris replied:
@pramodk You are right 👍 Actually I think that we should restrict the old `nrnivmodl-core` workflow like this `spec.satisfies('^coreneuron@0:0.16')`, so that in case there is a version named with a string like `develop`, `master` or `whatever` it is using the new workflow.

iomaganaris committed:
Better restriction for using old nrnivmodl-core arguments

iomaganaris committed:
Fixed get_cmake_args for the newer versions in coreneuron recipe

ferdonline commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -70,11 +70,14 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
+                '-i', include_flag, '-l', link_flag, '-n', self.mech_name,
+                '-v', str(spec.version), '-c', mods_location)
+        else:
+            which('nrnivmodl-core')('-i', include_flag, '-l', link_flag, mods_location)
```
Newer coreneurons dont accept versioning the library?

pramodk replied:
yeah, we discussed that offline.

pramodk commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -70,11 +70,14 @@ def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
+            which('nrnivmodl-core')('-i', include_flag, '-l', link_flag, mods_location)
         output_dir = spec.architecture.target
-        expected_name = "libcorenrnmech" + ('_' + self.mech_name if self.mech_name else '')
+        expected_name = "libcorenrnmech"
```
this won't work with `^coreneuron@0:0.16`? because we are still passing `'-v', str(spec.version)` for old version.

iomaganaris replied:
In the next line we call `find_libraries('libcorenrnmech*', output_dir)` which will look for all the files that start with `libcorenrnmech`. This means that even if there are libraries called `libcorenrnmech_neocortex.0.0.so` or in general that start with `libcorenrnmech` they will be found by the function `find_libraries`. I think that this is sufficient for both cases.

ohm314 replied:
@pramodk if this is OK for you can we merge this?

pramodk replied:
yup, sorry for missing this.

ferdonline commented:
Is there a substitute for the `-c` option? What is the default now?
edit: by what I saw, recent coreneuron never uses the internal mods, not even the built-in gap.mod, effectively as if `-c` was always true.
